network/tcp.go

```diff
@@ -265,7 +265,7 @@ type TCPListener struct {
 // the binding.
 // A subsequent call to Address() gives the actual listening
 // address which is different if you gave it a ":0"-address.
-func NewTCPListener(addr Address, s Suite) (*TCPListener, error) {
+func NewTCPListener(addr Address, s Suite, listenAddr Address) (*TCPListener, error) {
```

Comment:
There is a magical default for listenAddr, in this function. It must be documented.